optuna/integration/allennlp/_pruner.py

```diff
@@ -143,7 +81,7 @@ class AllenNLPPruningCallback(TrainerCallback):
 
     def __init__(
         self,
-        trial: Optional[optuna.trial.Trial] = None,
+        trial: optuna.trial.Trial,
```

Comment:
It could leave it as is but not giving `trial` doesn't work anymore after this PR.

The current implementation restores a `trial` from environment variables so `trial` is optional. But this logic will be completely removed. The only when `trial` is not passed to a pruning callback is that users uses the integration in the Jsonnet style usage. This means it always happens internally and I think the change of this doesn't affect users.

So, I want to make `trial` required. If you don't understand what I mean (my explanation must not be well), please ask me.